bee-protocol/src/workers/peer/manager_res.rs

```diff
@@ -96,6 +94,26 @@ impl PeerManager {
         self.0.read().peers.iter().for_each(|(id, (peer, _))| f(id, peer));
     }
 
+    pub(crate) fn find_first_fairly(&self, counter: &mut usize, f: impl Fn(&Peer) -> bool) -> Option<PeerId> {
```

Comment:
Do you think we can make the counter internal ? API would be cleaner.

Reply:
Actually I think it should be internal, it's the only way to guarantee 100% fairness.

Reply:
I will make it internal. I was thinking on removing the `first` as I think it is redundant. `Iterator::find` already stops on the first match

Reply:
Yeah I'd remove the `first` as well

Reply:
`fair_find` or ` find_fairly` ?

Reply:
whatever you prefer. I'd do `find_fairy` :trollface:

Reply:
Ahah, whatever, you choose

Reply:
🧚‍♂️